internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -397,28 +395,40 @@ func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Co
 		log.Infof("Setting default ironic agent image (%s) for infraEnv %s", ironicAgentImage, infraEnv.Name)
 	}
 
-	conf, err := ignition.GenerateIronicConfig(r.IronicServiceURL, r.IronicInspectorURL, *infraEnvInternal, ironicAgentImage)
+	ironicServiceURL, inspectorURL, err := r.BMOUtils.GetIronicServiceURLS()
```

Comment:
Looks like `r.BMOUtils.GetIronicServiceURLS()` will not return an error in case it failed to get the inspectionURL.
I'm pretty sure I fixed that already, perhaps this PR needs rebasing

Reply:
Rebased